Streamline backporting and improve backporting docs

joestringer opened the pull request:
Please review commit-by-commit.

Main functionality added in this PR is:

    backporting: Add summary log option to check-stable

    Allow writing a nicely-formatted PR message to a file:

    ```
        $ check-stable 1.4 my-pr.txt
        $ cat my-pr.txt
        v1.4 backports 2019-01-30

         * #xxxx -- commit title (@author)
         ...
    ```

See #6858 for an example. This file can be passed to [Github command-line tools](https://github.com/node-gh/gh) directly to open the backport PR from the commandline (shout out to @nebril for the suggestion!)

Also, serve `make render-docs` on `localhost:9080` as port `8080` conflicts with microk8s default ports, and document the new feature, as well as the feature in #6834.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6859)
<!-- Reviewable:end -->

joestringer committed:
backporting: Add summary log option to check-stable

Allow writing a nicely-formatted PR message to a file:

```
    $ check-stable 1.4 my-pr.txt
    $ cat my-pr.txt
    v1.4 backports 2019-01-30

     * #xxxx -- commit title (@author)
     ...
```

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
docs: Update backporting for the latest scripts

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
Makefile: Serve render-docs on port 9080.

When running `make render-docs`, serve it on port 9080 so it's less
likely to conflict with other local web servers running on port 8080.

Signed-off-by: Joe Stringer <joe@cilium.io>

joestringer committed:
docs: Fix backporting shell example formatting

Sphinx seems to expect indents of three spaces, make each indentation in
this section conform with this standard.

Signed-off-by: Joe Stringer <joe@cilium.io>

brb commented on Makefile:
```diff
@@ -300,8 +300,8 @@ docs-container:
 render-docs: test-docs
-	$(DOCKER) container run --rm -dit --name docs-cilium -p 8080:80 -v $$(pwd)/Documentation/_build/html/:/usr/local/apache2/htdocs/ httpd:2.4
-	@echo "$$(tput setaf 2)Running at http://localhost:8080$$(tput sgr0)"
+	$(DOCKER) container run --rm -dit --name docs-cilium -p 9080:80 -v $$(pwd)/Documentation/_build/html/:/usr/local/apache2/htdocs/ httpd:2.4
```
Could you update `Documentation/contributing.rst` to reflect the port change, as it currently says:

```
Now the documentation page should be browsable on http://localhost:8080.
```
?

joestringer replied:
Good catch.

brb reviewed (commented):
Nice work! Just one nit.

joestringer committed:
backporting: Add set-labels commands to check-stable

Output the full set-labels.py one-liner command when running
check-stable to assist backporters attempting to backport. It's up to
the backporter to exclude particular PRs if the PR is not included in
the PR.

Signed-off-by: Joe Stringer <joe@cilium.io>
